src/main/scala/akka/contrib/process/BlockingProcess.scala

```diff
@@ -102,17 +99,18 @@ class BlockingProcess(
 
   private val process = startProcess()
 
+  // This shutdown hook is required because the stdio actors will be blocked on a read. The only
+  // reliable means of unblocking them is to destroy the thing they're listening to.
+  context.actorOf(ShutdownHook.props(() => process.destroy()), "process-destroyer")
```

Comment:
@huntc

>That said, I think that the actor shutdown will occur in an orderly manner such that the process object is handled reasonably i.e. the parent actor will do nothing with the process object until its children have shutdown.

Is the order of shutdown of the children is guaranteed, so the process will never get destroyed -before- the reader and writer have been shut down?

>Any other thoughts or inspiration would be welcomed. Unfortunately the Java APIs do not provide a PID otherwise we could just kill it via the command line.

Right now there's also 2 different calls to `Process.destroy()` both when responding to `Destroy` message and when the shutdownhook runs its `postStop` call. Why multiple calls?

I don't see how it would be possible to write a new Process API that would be platform independent and also non-blocking as the JDK API is blocking.

Thoughts?